### PR TITLE
Polish translation of Tour of Scala: Higher - order functions 

### PR DESCRIPTION
**Pull request by kamilduda**
Add Polish translation for higher - order functions

### CODE REVIEW

**Comment by SethTisue** [comment]
review by @jstefanska?

File: _pl/tour/higher-order-functions.md [hunk 1]
```diff
@@ -6,38 +6,124 @@ discourse: false
...
 language: pl
 next-page: nested-functions
 previous-page: mixin-class-composition
 ---
 
-Scala pozwala na definiowanie funkcji wyższego rzędu. Są to funkcje, które przyjmują funkcje jako parametry lub których wynik jest też funkcją. Poniżej znajduje się przykład funkcji `apply`, która pobiera inną funkcję `f` i wartość `v` po to, by zwrócić wynik zastosowania `f` do `v`:
+Scala pozwala na definiowanie funkcji wyższego rzędu.
+Są to funkcje, które przyjmują funkcje jako parametry lub których wynik również jest funkcją. 
+Jest to możliwe, ponieważ w Scali funkcje są wartościami pierwszej kategorii (first-class values).
+Terminologia może w tym momencie wydawać się nie jasna, będzie używać pojęcia "funkcja wyższego rzędu" zarówno dla metod jak i funkcji przyjmujących jako parametry funkcje lub zwracających inne funkcje.
```
Comment (jstefanska) [inline _pl/tour/higher-order-functions.md]:
```suggestion
Terminologia może w tym momencie wydawać się niejasna, pojęcie "funkcja wyższego rzędu" będzie używane zarówno dla metod jak i funkcji przyjmujących jako parametry funkcje lub zwracających inne funkcje.
```

File: _pl/tour/higher-order-functions.md [hunk 2]
```diff
@@ -6,38 +6,124 @@ discourse: false
...
 ```tut
-def apply(f: Int => String, v: Int) = f(v)
+val salaries = Seq(20000, 70000, 40000)
+val doubleSalary = (x: Int) => x * 2
+val newSalaries = salaries.map(doubleSalary) // List(40000, 140000, 80000)
 ```
 
-_Uwaga: metody są automatycznie zamieniane na funkcje, jeżeli wymaga tego kontekst_
+Funkcja `doubleSalary` przyjmuje jako parametr wartość `x` typu `Int` i zwraca `x * 2`.
+W ogólności, krotka po lewej stronie strzałki `=>` jest listą parametrów, a wartość wyrażenia po prawej stronie jest tym, co zostanie zwrócone.
```
Comment (jstefanska) [inline _pl/tour/higher-order-functions.md]:
```suggestion
Ogólnie mówiąc, krotka po lewej stronie strzałki `=>` jest listą parametrów, a wartość wyrażenia po prawej stronie jest tym, co zostanie zwrócone.
```

File: _pl/tour/higher-order-functions.md [hunk 3]
```diff
@@ -6,38 +6,124 @@ discourse: false
...
+Aby zredukować trochę kod, możemy dodatkowo użyć funkcji anonimowej i przekazać ją bezpośrednio jako argument do funkcji `map`:
+
+```
+val salaries = Seq(20000, 70000, 40000)
+val newSalaries = salaries.map(x => x * 2) // List(40000, 140000, 80000)
+```
 
-Praktyczny przykład:
+Zauważ, że w powyższym przykładzie `x` nie jest zadeklarowane jako typ `Int`.
+Dzieje się tak, ponieważ kompilator może wywnioskować typ bazując na type funkcji oczekiwanej przez `map`.
```
Comment (jstefanska) [inline _pl/tour/higher-order-functions.md]:
```suggestion
Dzieje się tak, ponieważ kompilator może wywnioskować typ, bazując na typie funkcji oczekiwanej przez `map`.
```

**Comment by kamilduda** [comment]
@jstefanska done 😄

**Pull request merged**